src/Symfony/Component/Messenger/Transport/Schedule/Trigger/PeriodicalTrigger.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Schedule\Trigger;
+
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+
+final class PeriodicalTrigger implements TriggerInterface
+{
+    public function __construct(
+        private readonly int $intervalInSeconds,
+        private readonly \DateTimeImmutable $firstRun,
+        private readonly \DateTimeImmutable $priorTo,
+    ) {
+        if (0 >= $this->intervalInSeconds) {
+            throw new InvalidArgumentException('The `$intervalInSeconds` argument must be greater then zero.');
+        }
+    }
+
+    public static function create(
```

Comment:
It was. It's convenient to have a universal constructor from anything, but it might be difficult to understand how the constructor works and what to expect from it. It is nice to have a way to skip all "the magic" and just set properties.

Reply:
Shall we create some factory for triggers with methods like: "once", "hourly", "daily", etc?